generic_config_updater/change_applier.py

```diff
@@ -59,7 +59,12 @@ class ChangeApplier:
 
     updater_conf = None
 
-    def __init__(self):
+    def __init__(self, config_wrapper):
+        self.config_wrapper = config_wrapper
+        if type(self.config_wrapper) is DryRunConfigWrapper:
+            # No need to init the rest of the parameters
+            return
```

Comment:
In concept, there should be an interface, like `IChangeApplier` and both `ChangeApplier` and `ChangeDryRunApplier` implement the interface.

Suggest move new code into a new `class ChangeDryRunApplier` with a function `apply`.

The benefit is better performance in `apply` function, and better modularity.

Reply:
I was planning on modifying `ChaneApplier` itself to use `ConfigWrapper` for the low level calls. I think a new `DryRun` class for the `ChangeApplier` can give us better peace of mind about the future, especially if `ChangeApplier` impl changed in such a way that does not use the `ConfigWrapper`.

Reply:
updated